source/spirv_stats.cpp

```diff
@@ -34,49 +34,71 @@ using libspirv::SpirvStats;
 
 namespace {
 
+struct StatsContext {
+  SpirvStats* stats;
+
+  // Opcodes of already processed instructions. Used and updated in
+  // ProcessOpcode.
+  std::vector<uint32_t> opcodes;
```

Comment:
The comment must say how this vector is indexed.  When I first read this I assumed it was indexed by the opcode value.
But reading the rest of the code I think this vector contains the opcodes for instructions, in the order listed in the module. For example, opcodes[k] is the opcode for the instruction at position k (0-based).

Also, to avoid staleness in comments, we normally describe the data itself rather than where it is used.